src/server/auth/server/api_server.go

```diff
@@ -124,35 +130,122 @@ func NewAuthServer(env Env, public, requireNoncriticalServers, watchesEnabled bo
 }
 
 func (a *AuthServer) EnvBootstrap(ctx context.Context) error {
-	if a.env.Config.AuthRootToken == "" {
+	if !a.env.Config.ActivateAuth {
 		return nil
 	}
 	a.env.Logger.Info("Started to configure auth server via environment")
 	if err := func() error {
-		if _, err := a.Activate(ctx, &auth.ActivateRequest{
-			RootToken: a.env.Config.AuthRootToken,
-		}); err != nil {
-			if !errors.As(err, auth.ErrAlreadyActivated) {
+		ctx = internalauth.AsInternalUser(ctx, internalUser)
+		// handle auth activation
+		if a.env.Config.AuthRootToken != "" {
+			if _, err := a.Activate(ctx, &auth.ActivateRequest{
+				RootToken: a.env.Config.AuthRootToken,
+			}); err != nil {
+				if !errors.As(err, auth.ErrAlreadyActivated) {
+					return err
+				}
+				_, err := a.rotateRootToken(ctx,
+					&auth.RotateRootTokenRequest{
+						RootToken: a.env.Config.AuthRootToken,
+					})
 				return err
+			} else {
+				if a.env.Config.PachdSpecificConfiguration != nil {
+					if _, err := a.env.GetPfsServer().ActivateAuth(ctx, &pfs.ActivateAuthRequest{}); err != nil {
+						return errors.EnsureStack(err)
+					}
```

Comment:
What happens if the pod dies right here, i.e. PFS gets ActivateAuth called, but not PPS?

Reply:
yeah this is a bug. need to think about how to best handle this

Reply:
have a follow up PR to fix this: 
https://github.com/pachyderm/pachyderm/pull/7817/files